packages/gestalt/src/Link.js

```diff
@@ -96,6 +99,17 @@ const LinkWithForwardRef: AbstractComponent<Props, HTMLAnchorElement> = forwardR
     },
   );
 
+  // useOnNavigation is only accessible with Gestalt Provider
+  // and when onNavigation prop is passed to it
+  const { onNavigation } = useOnNavigation() || {};
+
+  const onNavigationClick =
```

Comment:
If the usage pattern is to get the onNavigation function and then call that with the href and onNavigationOptions, why not pass those to the custom hook?
```js
const onNavigationClick = useOnNavigation({ href, onNavigationOptions });
```
It'd be nice to adjust the return of useOnNavigation a bit as well, both to return the function directly (instead of needing to destructure it), and by returning a no-op instead of undefined, so later truthy checks aren't needed before calling the function.

Reply:
DONE